lib/multi.c

```diff
@@ -898,7 +898,7 @@ CURLMcode curl_multi_remove_handle(struct Curl_multi *multi,
   /* Remove the association between the connection and the handle */
   Curl_detach_connection(data);
 
-  if(data->set.connect_only && !data->multi_easy) {
+  if(data->set.connect_only && (data->multi != data->multi_easy)) {
```

Comment:
Why this change? If the easy interface is not used, the `multi_easy` pointer is NULL so isn't both the old and the new code achieving the same thing?

Reply:
> If the easy interface is not used, the `multi_easy` pointer is NULL

sometimes. the rub is if an easy handle is used with the easy interface and then the multi interface. in that case they point to different multis. this pr addresses that issue by prioritizing multi over multi_easy  because the former is always the in-use multi. my hang up with Curl_getconnectinfo (which is used in this logic) is, following this scenario, what happens when the easy handle is removed by the user via curl_multi_remove_handle, because that would seem to cause disconnection of connections in multi_easy and not multi. afaict, this change fixes a bug by checking that they are different and Curl_getconnectinfo prioritizes multi over multi_easy